donate/settings/base.py

```diff
@@ -0,0 +1,386 @@
+                'class': 'logging.StreamHandler'
+            },
+            'debug-error': {
+                'level': 'ERROR',
+                'filters': ['require_debug_true'],
+                'class': 'logging.StreamHandler'
+            },
+            'info': {
+                'level': 'INFO',
+                'class': 'logging.StreamHandler',
+                'formatter': 'verbose'
+            },
+            'rq_console': {
+                'level': 'DEBUG',
+                'class': 'rq.utils.ColorizingStreamHandler',
+                'formatter': 'rq_console',
+                'exclude': ['%(asctime)s']
+            },
+        },
+        'loggers': {
+            'django': {
+                'handlers': ['debug'],
+                'level': 'DEBUG'
+            },
+            'django.server': {
+                'handlers': ['debug'],
+                'level': 'DEBUG',
+            },
+            'django.request': {
+                'handlers': ['error'],
+                'propagate': False,
+                'level': 'ERROR'
+            },
+            'django.template': {
+                'handlers': ['debug-error'],
+                'level': 'ERROR'
+            },
+            'django.db.backends': {
+                'handlers': ['debug-error'],
+                'level': 'ERROR'
+            },
+            'django.utils.autoreload': {
+                'handlers': ['debug-error'],
+                'level': 'ERROR'
+            },
+            'rq.worker': {
+                'handlers': ['debug'],
+                'level': 'DEBUG',
+            },
+            'donate': {
+                'handlers': ['info'],
+                'level': 'INFO',
+            }
+        }
+    }
+
+    CSP_DEFAULT = (
+        '\'self\''
+    )
+
+    CSP_BASE_URI = env('CSP_BASE_URI', default=None)
+    CSP_CONNECT_SRC = env('CSP_CONNECT_SRC', default=None)
+    CSP_FRAME_ANCESTORS = env('CSP_FRAME_ANCESTORS', default=None)
+    CSP_FORM_ACTION = env('CSP_FORM_ACTION', default=None)
+    CSP_SANDBOX = env('CSP_SANDBOX', default=None)
+    CSP_REPORT_URI = env('CSP_REPORT_URI', default=None)
+    CSP_OBJECT_SRC = env('CSP_OBJECT_SRC', default=None)
+    CSP_MEDIA_SRC = env('CSP_MEDIA_SRC', default=None)
+    CSP_FRAME_SRC = env('CSP_FRAME_SRC', default=None)
+
+    @property
+    def CSP_DEFAULT_SRC(self):
+        return env('CSP_DEFAULT_SRC', default=self.CSP_DEFAULT)
+
+    @property
+    def CSP_SCRIPT_SRC(self):
+        return env('CSP_SCRIPT_SRC', default=self.CSP_DEFAULT)
+
+    @property
+    def CSP_IMG_SRC(self):
+        return env('CSP_IMG_SRC', default=self.CSP_DEFAULT)
+
+    @property
+    def CSP_FONT_SRC(self):
+        return env('CSP_FONT_SRC', default=self.CSP_DEFAULT)
+
+    @property
+    def CSP_STYLE_SRC(self):
+        return env('CSP_STYLE_SRC', default=self.CSP_DEFAULT)
+
+    @property
+    def CSP_WORKER_SRC(self):
+        return env('CSP_WORKER_SRC', default=self.CSP_DEFAULT)
+
+    @classmethod
```

Comment:
django-configurations provides three hooks for configuration classes, `pre_setup`, `setup`, and `post_setup`, where you can add code that needs to execute during the loading of your django settings.